packages/eas-cli/src/commands/env/push.ts

```diff
@@ -0,0 +1,176 @@
+      } else {
+        delete updateVariables[variable.name];
+      }
+    });
+
+    const existingDifferentSharedVariables = existingDifferentVariables.filter(
+      variable => variable.scope === EnvironmentVariableScope.Shared
+    );
+
+    if (existingDifferentSharedVariables.length > 0) {
+      const existingDifferentSharedVariablesNames = existingDifferentSharedVariables.map(
+        variable => variable.name
+      );
+      Log.error(
+        'Shared variables cannot be overwritten.\nPlease remove them from the env file or unlink them from the project:'
+      );
+      existingDifferentSharedVariablesNames.forEach(name => Log.error(`- ${name}`));
+      return;
+    }
+
+    if (existingDifferentVariables.length > 0) {
+      Log.warn('Some variables already exist in the environment.');
+
+      const { variablesToOverwrite } = await promptAsync({
+        type: 'multiselect',
+        name: 'variablesToOverwrite',
+        message: 'Select variables to overwrite:',
+        // @ts-expect-error property missing from `@types/prompts`
+        optionsPerPage: 20,
+        choices: existingDifferentVariables.map(variable => ({
+          title: `${variable.name}: ${updateVariables[variable.name].value}`,
+          value: variable.name,
+        })),
+      });
+
+      for (const existingVariable of existingVariables) {
+        const name = existingVariable.name;
+        if (variablesToOverwrite.includes(name)) {
+          updateVariables[name]['overwrite'] = true;
+        } else {
+          delete updateVariables[name];
+        }
+      }
+    }
+
+    const variablesToPush = Object.values(updateVariables);
+
+    if (variablesToPush.length === 0) {
+      Log.log('No new variables to push.');
+      return;
+    }
+
+    await EnvironmentVariableMutation.createBulkEnvironmentVariablesForAppAsync(
+      graphqlClient,
+      variablesToPush,
+      projectId
+    );
+
+    Log.log(`Uploaded env file to ${environment} environment.`);
+  }
+
+  private async parseEnvFileAsync(
+    envPath: string,
+    environment: string
+  ): Promise<Record<string, EnvironmentVariablePushInput>> {
+    if (!(await fs.exists(envPath))) {
+      throw new Error(`File ${envPath} does not exist.`);
+    }
+    const variables: Record<string, EnvironmentVariablePushInput> = {};
+    const content = await fs.readFile(envPath, 'utf8');
+
+    content
+      .trim()
+      .split('\n')
+      .filter(line => !line.startsWith('#') && line.includes('='))
+      .forEach(line => {
+        const [name, value] = line.split('=');
+        if (name && value && name.match(/^\w+$/)) {
+          const visibility = name.startsWith('EXPO_SECRET')
+            ? EnvironmentVariableVisibility.Sensitive
+            : EnvironmentVariableVisibility.Public;
+          variables[name] = {
+            name,
+            value,
+            environment,
+            visibility,
+          };
+        }
+      });
+    return variables;
+  }
```

Comment:
I would suggest using `dotenv.parse` for this

https://www.npmjs.com/package/dotenv#%EF%B8%8F-usage